### ISSUE TITLE
New-PSSession and SSH pass a password?

### ISSUE THREAD

**Issue by riahc3** [PowerShell/PowerShell#8904]
Im trying from Windows connecting to Linux thru SSH and there is a -Username parameter but I dont see a -Password parameter; This means that each time I connect, it asks for me a password.

Get-Credentials do not seem to work with -Hostname

**Comment by iSazonov** [comment]
Please read https://docs.microsoft.com/en-us/powershell/scripting/learn/remoting/ssh-remoting-in-powershell-core?view=powershell-6 and consider ssh-agent.

**Comment by riahc3** [comment]
> Please read https://docs.microsoft.com/en-us/powershell/scripting/learn/remoting/ssh-remoting-in-powershell-core?view=powershell-6 and consider ssh-agent.

I dont see anything about ssh-agent in the link you provided (searched)

And I dont see anything about passing a password (searched for password as well)

**Comment by chuanjiao10** [comment]
This means that each time I connect, it asks for me a password.
--- yes，you  can never save password use    **Clear text  or cipher text** on powershell ssh remoting.

1 use ssh key to do **no password ssh remoting**，It also has one benefit: verifying the client and server.
1.1 linux to do it:use ssh-copy-id
1.2 windows to do it: use my script ssh-copy-id3.ps1 
it is easy than linux,because the first time you do not need to enter a password.
it use winscp module.
#ssh copy id on windows.
```powershell
<#
脚本目的：
从win中复制ssh公钥，到目的ssh服务器。

前提条件：
install-module winscp

用法：
ssh-copy-id3.ps1 -目的ip 1.2.3.4
建议保存编码为：bom头 + utf8
#>

Param
(
	$目的ip = '192.168.1.2'
)

#先运行ssh-keygen回车，产生key文件。https://github.com/PowerShell/Win32-OpenSSH/releases
$key文件1 = "$env:USERPROFILE\.ssh\id_rsa.pub"
$key文件2 = "$env:USERPROFILE\.ssh\authorized_keys"
Copy-Item -LiteralPath $key文件1  -Destination $key文件2

$用户名 = 'root'
$用户密码明文 = '这里填入你的ssh密码明文。'
$用户密码密文 = ConvertTo-SecureString $用户密码明文 -AsPlainText -Force
$我的登陆凭据 = New-Object System.Management.Automation.PSCredential ($用户名,$用户密码密文)
#QQ群号=183173532,名称=powershell交流群，2018-12-26
$sftp连接参数 = new-WinSCPSessionOption -Protocol Sftp -HostName $目的ip  -Credential  $我的登陆凭据
$指纹 = Get-WinSCPHostKeyFingerprint -SessionOption $sftp连接参数
$sftp连接参数.SshHostKeyFingerprint = $指纹
$sftp连接 = new-WinSCPSession -SessionOption $sftp连接参数

if (Test-WinSCPPath -Path '/root/.ssh' -WinSCPSession $sftp连接)
{
	Remove-WinSCPItem -Path '/root/.ssh' -Confirm:$false -WinSCPSession $sftp连接
}

$权限700 = New-WinSCPTransferOption -FilePermissions (New-WinSCPItemPermission -Octal 700)
New-WinSCPItem -Path '/root/.ssh' -ItemType Directory -TransferOptions  $权限700 -WinSCPSession $sftp连接

$权限600 = New-WinSCPTransferOption -FilePermissions (New-WinSCPItemPermission -Octal 600)
Send-WinSCPItem -LocalPath $key文件2 -RemotePath '/root/.ssh/' -TransferOptions $权限600 -WinSCPSession $sftp连接

Remove-WinSCPSession -WinSCPSession $sftp连接



```

**Comment by riahc3** [comment]
> This means that each time I connect, it asks for me a password.
> --- yes，you can never save password use **Clear text or cipher text** on powershell ssh remoting.
> The powershell team considers it to be a low security level and prohibits this.
> 

This is not the Powershell team's call; Of course the recommended method is using a key but some devices do not support this. Plain text credentials should always be supported. Period. 

How is this BASIC feature available in every single SSH client known and unknown missing in Powershell?

**Comment by rkeithhill** [comment]
You don't have to use New-PSSession.  You can use ssh.exe "directly" with plain text credentials and then once you're on the remote device, fire up `pwsh`.  Yeah, I know - not quite the same functionality as PS remoting.  But I log into remote Linux devices by ssh'ing into them and then firing up pwsh *far* more than I use *-PSSession.

**Comment by chuanjiao10** [comment]
@riahc3  Are you a hacker? Powershell team Be careful that hackers like you are right.

**Comment by riahc3** [comment]
@rkeithhill What you are proposing is NOT Powershell; You are mixing it with a external program. 

I dont understand WHY this isnt implemented. Its something pretty basic and simple. There are some devices that simply DO NOT support passwordless SSH.

Please implement this into the cmdlet. Even if you require a SecureString

**Comment by rkeithhill** [comment]
SecureString does not encrypt on macOS or Linux.

**Comment by iSazonov** [comment]
As a note, MSFT announced plans to resolve this in 7.0 milestone somehow.

**Comment by nikonthethird** [comment]
@iSazonov has this been resolved in 7.0?
